src/ansi.rs

```diff
@@ -672,6 +673,70 @@ pub trait Handler {
 
     /// Set mouse cursor icon.
     fn set_mouse_cursor_icon(&mut self, _: CursorIcon) {}
+
+    /// Report current keyboard mode.
+    ///
+    /// ## Caveat
+    ///
+    /// This request must be done successively with the [`identify_terminal`] request, given
+    /// clients rely on the later to identify whether the terminal suppors kitty's keyboard
+    /// protocol.
+    ///
+    /// [`identify_terminal`]: Self::identify_terminal
+    fn report_keyboard_mode(&mut self) {}
+
+    /// Push keyboard mode into the keyboard mode stack.
+    fn push_keyboard_mode(&mut self, _mode: KeyboardMode) {}
+
+    /// Pop the given amount of keyboard modes from the
+    /// keyboard mode stack.
+    fn pop_num_keyboard_modes(&mut self, _to_pop: u16) {}
```

Comment:
```suggestion
    /// Pop the given amount of keyboard modes from the
    /// keyboard mode stack.
    fn pop_keyboard_modes(&mut self, _to_pop: u16) {}
```

I think the separation between "mode" and "modes" should be enough to signify multiple are being popped. The name is a bit awkward and long otherwise and I don't think there's ever going to be a chance of confusion anyway.